ReactiveXComponent/Common/ThreadSafeList.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+
+namespace ReactiveXComponent.Common
+{
+    public class ThreadSafeList<T> : IList<T>
+    {
+        private readonly List<T> _list = new List<T>();
+        private readonly object _lock = new object();
+
+        public IEnumerator<T> GetEnumerator()
+        {
+            lock (_lock)
+            {
+                return _list.GetEnumerator();
+            }
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        public void Add(T item)
+        {
+            lock (_lock)
+            {
+                _list.Add(item);
+            }
+        }
+
+        public void Clear()
+        {
+            lock (_lock)
+            {
+                _list.Clear();
+            }
+        }
+
+        public bool Contains(T item)
+        {
+            lock (_lock)
+            {
+                return _list != null && _list.Contains(item);
+            }
+        }
+
+        public void CopyTo(T[] array, int arrayIndex)
+        {
+            lock (_lock)
+            {
+                _list.CopyTo(array, arrayIndex);
+            }
+        }
+
+        public bool Remove(T item)
+        {
+            lock (_lock)
+            {
+                return _list != null && _list.Remove(item);
+            }
+        }
+
+        public int Count => _list.Count;
+
+        public bool IsReadOnly { get; }
```

Comment:
The ThreadSafeList isn't a  ReadOnly List, this is why the IsReadOnly flag is never assigned.